storage.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"io"
+	"net/http"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/net/context"
+)
+
+var (
+	client     *storage.Client
+	bucket     *storage.BucketHandle
+	bucketName string
+)
+
+func init() {
+	var err error
+	client, err = storage.NewClient(context.Background())
+	if err != nil {
+		panic(err)
+	}
+
+	bucketName = os.Getenv("SITE_BUCKET")
+	if bucketName == "" {
+		panic("No SITE_BUCKET given")
+	}
+}
+
+func handleStaticRequest(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Served-With-Love-By", "Gophers")
+
+	bucket := client.Bucket(bucketName)
+
+	path := r.URL.Path[1:] // removing the leading / here
+
+	if path == "" {
+		path = "index.html"
+	}
+
+	if path[:7] == "assets/" {
+		w.Header().Set("Cache-Control", "max-age=86400")
+	}
```

Comment:
all of this path-specific control flow is exactly why http routers are so awesome! b/c the router can do all this control flow for you :D

try maybe, (in your main)

```
http.Handle("/assets", assetHandler) // assetHandler is a http.HandlerFunc (like this func), and just set the header for all requests at the top of assetHandler func body :)
http.Handle("/", indexHandler) // least-specific route last
```